haystack/file_converter/base.py

```diff
@@ -89,3 +89,24 @@ def run(self, file_path: Path, meta: Optional[Dict[str, str]] = None, remove_num
 
         result = {"document": document, **kwargs}
         return result, "output_1"
+
+
+class FileTypeClassifier(BaseComponent):
+    """
+    Route files in an Indexing Pipeline to corresponding file converters.
+    """
+    outgoing_edges = 4
+
+    def run(self, file_path: Path, **kwargs):  # type: ignore
```

Comment:
How about this simplification -
```
def run(self, file_path: Path, **kwargs):  # type: ignore
        output = {"file_path": file_path, **kwargs}
        ext = file_path.name.split(".")[-1].lower()
        try:
                 index = ["txt", "pdf", "md", "docx"].index(ext) + 1
                 return output, f"output_{index}"
        except ValueError:
            raise Exception(f"Files with an extension '{ext}' are not supported.")
```

Also we can also support html as tika support html to txt conversion.